### PR TITLE
Using StoreMixin for ChronosStore

### PR DESCRIPTION
**Pull request by mlunoe**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @kennyt

File: src/js/stores/ChronosStore.js [hunk 1]
```diff
@@ -16,8 +16,6 @@ import {
   SERVER_ACTION
 } from '../constants/ActionTypes';
 
-import VisibilityStore from './VisibilityStore';
```
Comment (jfurrow) [inline src/js/stores/ChronosStore.js]: We still use `VisibilityStore` in here to check visibility

Comment (mlunoe) [inline src/js/stores/ChronosStore.js]: I was too fast 🏇

**Comment by ssk2** [comment]
Retest this please.

**Comment by ssk2** [comment]
retest this please

**Comment by ssk2** [comment]
retest this please

**Comment by ssk2** [comment]
retest this please

File: src/js/stores/ChronosStore.js [hunk 2]
```diff
@@ -66,10 +66,7 @@ class ChronosStore extends EventEmitter {
       return true;
     });
 
-    VisibilityStore.addChangeListener(
-      VISIBILITY_CHANGE,
-      this.onVisibilityStoreChange.bind(this)
-    );
+    this.store_initializeListeners([{name: 'visibility', events: ['change']}]);
```
Comment (orlandohohmeier) [inline src/js/stores/ChronosStore.js]: Do we really wan't to introduce the store mixin only to add a listener to the visibility store? I thought our overall plan is to "remove" mixins...  I'd rather introduce something like the recurrent task util (https://github.com/dcos/dcos-ui/pull/91) to handle this in a clean consistent way. What do you think?

**Comment by orlandohohmeier** [comment]
Do we really want to introduce the store mixin only to add a listener to the visibility store? I thought our overall plan is to "remove" mixins...  I'd rather introduce something like the recurrent task util (https://github.com/dcos/dcos-ui/pull/91) to handle this in a clean consistent way. What do you think?

**Comment by kennyt** [comment]
@orlandohohmeier i thought we agreed to have some mixins. like the `StoreMixin`. we have many stores already using the `StoreMixin` to listen to other stores atm (like this PR)

**Comment by orlandohohmeier** [comment]
@kennyt Yeah, sorry my comment was misleading. I meant, that we agreed (if I recall correctly) to be cautions with the usage and introduction of mixins and that we consider the alternatives before introducing new ones. 
If we want to move towards this store pattern, let's make sure that we have a rollout plan in place and discuss this before introduce it, as there aren't any stores that use the store mixin, at least as far as I can tell. 

/cc @dcos/frontend

**Comment by kennyt** [comment]
@orlandohohmeier apologies, all of the stores that use `StoreMixin` are inside of the private plugins. no store in dcos-ui uses it because there's been no need to until now i suppose

**Comment by rcorral** [comment]
Hey guys, you're both right! Let's have a rollout plan for this as there's many others stores that can benefit from this!

**Comment by mlunoe** [comment]
This change is global. The ones in the plugins have already landed

**Comment by mlunoe** [comment]
Guys, since change is already in the plugins, could we get this in, and then make a decision whether we want to change it? @rcorral @orlandohohmeier

**Comment by orlandohohmeier** [comment]
@mlunoe Do we need this now? Otherwise I'd suggest to wait with this adjustment and the needed discussion (which might cost use more time than we want) until we've implemented all the features for 1.8. What do you think?

**Comment by rcorral** [comment]
I agree with @orlandohohmeier we should wait on this rather than adding more inconsistencies. It's not something that we need, but an improvement. Let's rolls this out globally later and let's plan for it so it doesn't add additional unplanned work.

**Comment by MatApple** [comment]
@orlandohohmeier I actually really like the Util you proposed.

**Comment by MatApple** [comment]
Lets close this for another day.

**Pull request closed without merging**